e2e/tests/test_e2e.py

```diff
@@ -0,0 +1,327 @@
+        k8s.api.core_v1.create_namespace(v1_namespace)
+
+        # submit the most recent operator image built on the Docker host
+        with open("manifests/postgres-operator.yaml", 'r+') as f:
+            operator_deployment = yaml.safe_load(f)
+            operator_deployment["spec"]["template"]["spec"]["containers"][0]["image"] = os.environ['OPERATOR_IMAGE']
+            yaml.dump(operator_deployment, f, Dumper=yaml.Dumper)
+
+        for filename in ["operator-service-account-rbac.yaml",
+                         "configmap.yaml",
+                         "postgres-operator.yaml"]:
+            k8s.create_with_kubectl("manifests/" + filename)
+
+        k8s.wait_for_operator_pod_start()
+
+        actual_operator_image = k8s.api.core_v1.list_namespaced_pod(
+            'default', label_selector='name=postgres-operator').items[0].spec.containers[0].image
+        print("Tested operator image: {}".format(actual_operator_image))  # shows up after tests finish
+
+        k8s.create_with_kubectl("manifests/minimal-postgres-manifest.yaml")
+        k8s.wait_for_pod_start('spilo-role=master')
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_multi_namespace_support(self):
+        '''
+        Create a customized Postgres cluster in a non-default namespace.
+        '''
+        k8s = self.k8s
+
+        with open("manifests/complete-postgres-manifest.yaml", 'r+') as f:
+            pg_manifest = yaml.safe_load(f)
+            pg_manifest["metadata"]["namespace"] = self.namespace
+            yaml.dump(pg_manifest, f, Dumper=yaml.Dumper)
+
+        k8s.create_with_kubectl("manifests/complete-postgres-manifest.yaml")
+        k8s.wait_for_pod_start("spilo-role=master", self.namespace)
+        self.assert_master_is_unique(self.namespace, version="acid-test-cluster")
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_scaling(self):
+        """
+           Scale up from 2 to 3 and back to 2 pods by updating the Postgres manifest at runtime.
+        """
+
+        k8s = self.k8s
+        labels = "version=acid-minimal-cluster"
+
+        k8s.wait_for_pg_to_scale(3)
+        self.assertEqual(3, k8s.count_pods_with_label(labels))
+        self.assert_master_is_unique()
+
+        k8s.wait_for_pg_to_scale(2)
+        self.assertEqual(2, k8s.count_pods_with_label(labels))
+        self.assert_master_is_unique()
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_taint_based_eviction(self):
+        """
+           Add taint "postgres=:NoExecute" to node with master. This must cause a failover.
+        """
+        k8s = self.k8s
+        labels = 'version=acid-minimal-cluster'
+
+        # get nodes of master and replica(s) (expected target of new master)
+        current_master_node, failover_targets = k8s.get_spilo_nodes(labels)
+        num_replicas = len(failover_targets)
+
+        # if all pods live on the same node, failover will happen to other worker(s)
```

Comment:
i'd expand this  comment to better explain why we can have `len(failover_targets) == 0`
1. single pod clusters (e.g. in test environments) - a new master pod must be started on another worker node meaning longer downtimes for such clusters
2. clusters that have master/replica on the same node due to unused pod anti-affinity -> a new replica must be started on another worker node for  a failover to take place. 
3. other clusters -> failover takes place / `failover_targets` not empty